lib/dialects/postgres/query-generator.js

```diff
@@ -564,24 +569,32 @@ const QueryGenerator = {
     return this.quoteIdentifier(Utils.removeTicks(this.escape(val), "'"));
   },
 
+
+
   expandFunctionParamList(params) {
     if (Utils._.isUndefined(params) || !Utils._.isArray(params)) {
       throw new Error('expandFunctionParamList: function parameters array required, including an empty one for no arguments');
     }
 
-    const paramList = Utils._.each(params, curParam => {
+    const paramList = [];
+    Utils._.each(params, curParam => {
       const paramDef = [];
       if (Utils._.has(curParam, 'type')) {
         if (Utils._.has(curParam, 'direction')) { paramDef.push(curParam.direction); }
         if (Utils._.has(curParam, 'name')) { paramDef.push(curParam.name); }
         paramDef.push(curParam.type);
       } else {
-        throw new Error('createFunction called with a parameter with no type');
+        throw new Error('expandFunctionParamList: parameter missing type');
```

Comment:
We don't need to change this error message, `expandFunctionParamList` is a private api that is used by `createFunction`.